django/core/cache/backends/redis.py

```diff
@@ -0,0 +1,239 @@
+"""Redis cache backend."""
+
+import pickle
+import random
+import re
+import urllib
+
+import redis
+
+from django.core.cache.backends.base import DEFAULT_TIMEOUT, BaseCache
+from django.core.exceptions import ImproperlyConfigured
+from django.utils.functional import cached_property
+from django.utils.module_loading import import_string
+
+
+class BaseCacheSerializer:
+    def dumps(self):
+        raise NotImplementedError
+
+    def loads(self):
+        raise NotImplementedError
+
+
+class PickleCacheSerializer(BaseCacheSerializer):
+    def __init__(self, options):
+        self._pickle_version = -1
+
+        if 'PICKLE_VERSION' in options:
+            try:
+                self._pickle_version = int(options['PICKLE_VERSION'])
+            except (ValueError, TypeError):
+                raise ImproperlyConfigured('must be an integer')
+
+    def dumps(self, value):
+        return pickle.dumps(value, self._pickle_version)
+
+    def loads(self, value):
+        return pickle.loads(value)
+
+
+class RedisCacheClient:
+    def __init__(self, servers, options):
+        self._servers = servers
+        self._pools = [None] * len(servers)
+        self._options = options
+
+        self._pool_class = redis.ConnectionPool
+        self._client = redis.Redis
+        self._serializer = PickleCacheSerializer(self._options)
```

Comment:
And this would be...
```suggestion
        self._serializer = PickleSerializer(pickle_protocol)
```